Assets/Scripts/Loop/TimerManager.cs

```diff
@@ -0,0 +1,121 @@
+/******************************************************************
+*    Author: Elijah Vroman
+*    Contributors: Elijah Vroman
+*    Date Created: 5/30/24 (Yes, very late...) 
+*    Description: Help ive spent far too long making a flexible timer
+*    manager ahhhhhhhhhh
+*    Anyways, this things main idea it to have a list of Timer classes
+*    that you can affect if you have a bunch of timers running around. 
+*    If programmers want to make a timer, instead of making their own 
+*    they can just use the methods from this script 
+*    The timer class has an "update" method but thats just what this calls every
+*    update. All the bs in update here is because the way the stacks
+*    were running cause of the event it would try to delete a timer 
+*    while the collection was still being iterated upon. Because we do need
+*    an event specific to the timer, i had to look for solutions.
+*******************************************************************/
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+[Serializable]
+public struct TimerStruct
+{
+    public string timerName;
+    public Timer timer;
+
+    public TimerStruct(string name, int durationInSeconds)
+    {
+        timerName = name;
+        timer = new Timer(durationInSeconds);
+    }
+}
+public class TimerManager : MonoBehaviour
+{
+    #region Instance
+    //regions are cool, i guess. Just hiding boring stuff
+    public static TimerManager Instance { get; private set; }
+
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this.gameObject);
+        }
+        else
+        {
+            Instance = this;
+            DontDestroyOnLoad(this.gameObject);
+        }
+    }
+    #endregion
+
+    [SerializeField] private List<TimerStruct> timers = new List<TimerStruct>();
```

Comment:
Move this to the start of the class

Reply:
Is there a reason for this to be serialized? Write up some documentation on the timers separately so the design team knows how to make their own timers as needed. Unless you can make a timer with this field, it should be private. [Making this a trello task]

Reply:
> Is there a reason for this to be serialized? Write up some documentation on the timers separately so the design team knows how to make their own timers as needed. Unless you can make a timer with this field, it should be private. [Making this a trello task]

Good point, it probably doesnt need to be since timers can only be seen in that list during runtime; not very helpful for devs. Also timers arent really for the game designers I kind of just meant it as QOL for coders if they ever wanted one and didnt want to write up a coroutine